src/main/java/de/fraunhofer/aisec/cpg/frontends/cpp/DeclaratorHandler.java

```diff
@@ -219,6 +221,22 @@ private RecordDeclaration handleCompositeTypeSpecifier(CPPASTCompositeTypeSpecif
     lang.getScopeManager().enterScope(recordDeclaration);
     lang.getScopeManager().addValueDeclaration(recordDeclaration.getThis());
 
+    handleMembers(ctx, recordDeclaration);
+
+    if (recordDeclaration.getConstructors().isEmpty()) {
+      de.fraunhofer.aisec.cpg.graph.ConstructorDeclaration constructorDeclaration =
+          NodeBuilder.newConstructorDeclaration(
+              recordDeclaration.getName(), recordDeclaration.getName(), recordDeclaration);
+      recordDeclaration.getConstructors().add(constructorDeclaration);
+      lang.getScopeManager().addValueDeclaration(constructorDeclaration);
+    }
+
+    lang.getScopeManager().leaveScope(recordDeclaration);
+    return recordDeclaration;
+  }
+
+  private void handleMembers(
```

Comment:
This `handle` function brakes the pattern of returning the handled type, maybe rename this to avoid confusion.